### PR TITLE
Change BrotliCompress response files to quote file paths

### LINKED ISSUE #26061: Failure to publish WASM when project source path contains blank spaces - Post-upgrade VS 2022 to 17.2.4
Since upgrade of Visual Studio 2022 from 17.2.3 to 17.2.4 on june 15th, I cannot publish any Blazor Web Application project anymore.

This goes for new and existing projects. Even a simple publish to folder fails with error message:

`C:\Program Files\dotnet\sdk\6.0.301\Sdks\Microsoft.NET.Sdk.BlazorWebAssembly\targets\Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets(614,5): Error MSB6006: "dotnet.exe" exited with code 1.
`

After hours of frustating attempts, I found that it was related to the location of my projects. **Parent path to my dev folder contained blank spaces!**

Changing VS MSBuild output verbosity, I found that the error was related to Blazor WebAssembly publish artifacts compression task:

```
...
Compressing Blazor WebAssembly publish artifacts. This may take a while...
Creating directory "obj\Release\net6.0\compress\".
C:\Program Files\dotnet\dotnet.exe "C:\Program Files\dotnet\sdk\6.0.301\Sdks\Microsoft.NET.Sdk.BlazorWebAssembly\targets\..\tools\net6.0\Microsoft.NET.Sdk.BlazorWebAssembly.Tool.dll" brotli
-s
C:\Dev\With Space\BlazorAppSol\BlazorApp\Client\bin\Release\net6.0\wwwroot\_framework\blazor.webassembly.js
-o
C:\Dev\With Space\BlazorAppSol\BlazorApp\Client\obj\Release\net6.0\compress\N6++qccl.br
...
Unrecognized command or argument 'Space\BlazorAppSol\BlazorApp\Client\bin\Release\net6.0\wwwroot\_framework\blazor.webassembly.js'.
...
C:\Program Files\dotnet\sdk\6.0.301\Sdks\Microsoft.NET.Sdk.BlazorWebAssembly\targets\Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets(614,5): Error MSB6006: "dotnet.exe" exited with code 1.
```

The above output was produced by creating a new Blazor Web Assembly project under a root folder with path 'C:\Dev\With Space\' and publishing it to relative folder 'bin\Release\net6.0\publish\'.

Is this a known issue?

Configuration:
Windows 10 Enterprise 21H2 Build 19044.1766
Visual Studio Enterprise 2022 64-bit version 17.2.4
Blazor WebAssembly App project with .NET 6 and ASP.NET Core host

### PR DESCRIPTION
**Pull request by baronfel**
## Description

Fixes a regression in the parsing/handling of response files (raised in #26061) by ensuring that file paths are quoted in response files generated by the brotli compression tool task. This ensures that the response files can be read by System.CommandLine, which updated and brought along this change in handling.

Given that this utility is the only report of this problem, we decided this was an acceptable fix for the issue. If it proves to be more widespread we can reimplement the old tokenization inside of Parser.cs. 

If/when the other razor CLI tools migrate to System.CommandLine we'll need to do the same quoting of file paths.

## Customer Impact

Users using the brotli compression feature cannot compile their applications if the application is in a path containing spaces. The user must rename containing folders to work around this at the moment.

## Regression

Yes. Introduced in 6.0.302.

## Testing

New tests verifying the contents of the generated response files were added for the Brotli Compression task.

## Risk

The risk of taking this is low - it only impacts response file generation for the Brotli compression command, which is also authored in the SDK and not distributed anywhere else.

### CODE REVIEW

**Comment by baronfel** [comment]
This is targeted for the next 6.0.3xx servicing release to fix a regression, so it will need servicing consideration, but I still need review from code owners.

**Review by javiercn — APPROVED** [review]
Brotli changes look good.

I can't speak about the NU1505 changes.

File: src/BlazorWasmSdk/Tasks/BrotliCompress.cs
Comment (baronfel) [reply src/BlazorWasmSdk/Tasks/BrotliCompress.cs:116]: That's correct - S.CL changed the logic (to something that's more reasonable cross-platform) and this just happened to trigger it. Longer-term resolutions for this might include:

* not doing this tool-based execution at all in this task, and instead doing the logic in _code_ in the task (unsure if this is intended or not?)
* MSBuild ToolTask helpers/support for this?

**Comment by baronfel** [comment]
The NU1505 changes have also been validated by @wtgodbe (we think the build environment for full framework got ahead of us). We won't take those changes in 6.0.4xx and 7.0.1xx, since the core SDK problem has been fixed in those branches.

**Comment by wtgodbe** [comment]
https://github.com/dotnet/sdk/pull/26133 covered many of the NU1505 changes, may be worth rebasing

File: src/BlazorWasmSdk/Tasks/BrotliCompress.cs
Comment (javiercn) [reply src/BlazorWasmSdk/Tasks/BrotliCompress.cs:116]: We can't do the brotli compression on a task because it needs to support net462 and brotli is not available there :(

Comment (baronfel) [reply src/BlazorWasmSdk/Tasks/BrotliCompress.cs:116]: Aha! I thought it might have been something like this. Thanks for clarifying.

**Comment by baronfel** [comment]
Solid point, I'll do that now.

**Comment by baronfel** [comment]
Closing this as we are going to revert the S.CL update now. Additional reports of the regression in 6.0.3xx have triggered this.

**Pull request closed without merging**
